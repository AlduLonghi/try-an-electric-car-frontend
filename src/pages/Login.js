import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setUser, loggedIn } from '../redux/actions/user';

const Login = ({ setUser, loggedIn }) => {
  const history = useHistory();
  const [loginInputs, setLoginInputs] = useState({ email: '', password: '' });

  const handleChange = e => {
    setLoginInputs({ ...loginInputs, [e.target.name]: e.target.value });
  };

  const handleOnClick = () => {
    console.log(loginInputs);
    fetch('http://localhost:3000/login', {
      method: 'POST',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(loginInputs),
    })
      .then(res => {
        setUser(res.data);
        loggedIn('true');
        history.push('/home');
      }).catch(() => loggedIn('false'));
  };

  return (
    <div className="login-cont d-flex align-items-center">
      <form className="login-form">
        <div className="form-group">
          <input type="email" className="form-control" name="email" placeholder="Enter email" onChange={handleChange} />
          <small id="emailHelp" className="form-text text-muted">Well never share your email with anyone else.</small>
        </div>
        <div className="form-group">
          <input type="password" className="form-control" name="password" placeholder="Password" onChange={handleChange} />
        </div>
        <button type="button" className="btn btn-primary" onClick={handleOnClick}>Login</button>
      </form>
    </div>
  );
};

Login.propTypes = {
  setUser: PropTypes.func.isRequired,
  loggedIn: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  setUser,
  loggedIn,
};

export default connect(null, mapDispatchToProps)(Login);
