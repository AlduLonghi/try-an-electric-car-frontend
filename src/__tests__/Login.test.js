import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Login from '../pages/Login';
import store from '../redux/store';

it('renders Login page correctly', () => {
    const tree = renderer.create(
        <Provider store={store}>
          <Login />
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});