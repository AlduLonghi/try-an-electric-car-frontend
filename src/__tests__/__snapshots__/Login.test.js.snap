// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page correctly 1`] = `
<div
  className="auth-cont d-flex align-items-center justify-content-center py2"
>
  <div
    className="row auth-row mx-0"
  >
    <div
      className="col-md-9 d-flex align-items-center"
    >
      <form
        className="auth-form"
      >
        <p
          className="form-title"
        >
          Welcome back!
        </p>
        <div>
          <span
            className="text-muted text-uppercase mb-3"
          >
            Email
          </span>
          <input
            className="input"
            name="email"
            onChange={[Function]}
            type="email"
          />
        </div>
        <div
          className=""
        >
          <span
            className="text-muted text-uppercase mb-3"
          >
            Password
          </span>
          <input
            className="input"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
        <button
          className="d-block mx-auto py-2 text-uppercase text-center"
          onClick={[Function]}
          type="button"
        >
          Log in
        </button>
      </form>
    </div>
    <div
      className="col-md-3 auth-side"
    >
      <div>
        <p
          className="mb-3"
        >
          New here?
        </p>
        <p>
          Sign up and discover the wonderful universe of electric cars
        </p>
      </div>
      <button
        className="side-button d-block text-uppercase py-2 px-4 mx-auto"
        onClick={[Function]}
        type="button"
      >
        Sign up
      </button>
    </div>
  </div>
</div>
`;
